src/lib/swapper/swappers/LifiSwapper/LifiSwapper2.ts

```diff
@@ -0,0 +1,130 @@
+import { getUnsignedTx } from './utils/getUnsignedTx/getUnsignedTx'
+
+const executedTrades: Map<string, GetStatusRequest> = new Map()
+const tradeQuoteMetadata: Map<string, Route> = new Map()
+
+let lifiChainMapPromise: Promise<Result<Map<ChainId, ChainKey>, SwapErrorRight>> | undefined
+
+export const lifi: Swapper2 = {
+  getTradeQuote: async (
+    input: GetTradeQuoteInput,
+    assets: Partial<Record<AssetId, Asset>>,
+    sellAssetPriceUsdPrecision: string,
+  ): Promise<Result<TradeQuote2, SwapErrorRight>> => {
+    if (lifiChainMapPromise === undefined) lifiChainMapPromise = getLifiChainMap()
+
+    const maybeLifiChainMap = await lifiChainMapPromise
+
+    if (maybeLifiChainMap.isErr()) return Err(maybeLifiChainMap.unwrapErr())
+
+    const tradeQuoteResult = await getTradeQuote(
+      input as GetEvmTradeQuoteInput,
+      maybeLifiChainMap.unwrap(),
+      assets,
+      sellAssetPriceUsdPrecision,
+    )
+
+    return tradeQuoteResult.map(({ selectedLifiRoute, ...tradeQuote }) => {
+      const { receiveAddress, affiliateBps } = input
+
+      // TODO: quotes below the minimum arent valid and should not be processed as such
+      // selectedLifiRoute willbe missing for quotes below the minimum
+      if (!selectedLifiRoute) throw Error('missing selectedLifiRoute')
+
+      const id = selectedLifiRoute.id
+
+      // store the lifi quote metadata for transaction building later
+      tradeQuoteMetadata.set(id, selectedLifiRoute)
+
+      return { id, receiveAddress, affiliateBps, ...tradeQuote }
+    })
+  },
+
+  getUnsignedTx: async ({ from, tradeQuote, stepIndex }: GetUnsignedTxArgs): Promise<ETHSignTx> => {
+    const lifiRoute = tradeQuoteMetadata.get(tradeQuote.id)
+
+    if (!lifiRoute) throw Error('missing trade quote metadata')
+
+    const { accountNumber, sellAsset } = tradeQuote.steps[stepIndex]
+
+    const unsignedTx = await getUnsignedTx({
+      lifiStep: lifiRoute.steps[stepIndex],
+      accountNumber,
+      sellAsset,
+      // discriminated union between from or xpub in GetUnsignedTxArgs, but since Li.Fi only deals with EVMs, from is always going to be defined
+      from: from!,
+    })
+
+    return unsignedTx
+  },
+
+  executeTrade: async ({ txToExecute, wallet, chainId }: ExecuteTradeArgs) => {
+    const adapterManager = getChainAdapterManager()
+    const adapter = adapterManager.get(chainId) as unknown as ChainAdapter<EvmChainId>
+
+    if (wallet.supportsOfflineSigning()) {
+      const signedTx = await adapter.signTransaction({
+        txToSign: txToExecute as ETHSignTx,
+        wallet,
+      })
+
+      const txid = await adapter.broadcastTransaction(signedTx)
+
+      return txid
+    }
+
+    if (wallet.supportsBroadcast() && adapter.signAndBroadcastTransaction) {
+      const txid = await adapter.signAndBroadcastTransaction?.({
+        txToSign: txToExecute as ETHSignTx,
+        wallet,
+      })
+
+      return txid
+    }
+
+    throw Error('sign and broadcast failed')
+  },
```

Comment:
Not required as part of this PR and can have it as a follow-up but we could use the `broadcast` function:

https://github.com/shapeshift/web/blob/1b8b2449619c00108c6857c2baed70e4d0b3e16d/src/lib/utils/evm/index.ts#L138-L151

Despite the name, it actually *signs* and broadcasts and does the exact same checks as here